pkg/controller/migration/convert/ippools.go

```diff
@@ -114,6 +114,33 @@ func handleIPPools(c *components, install *operatorv1.Installation) error {
 		}
 	}
 
+	// If IPv6 only and bird backend is used, check that CALICO_ROUTER_ID, if defined, is set to `hash`.
+	// Custom router ID values are only used for manual calico-node deployments and not
+	// applicable to calico-node running as a daemonset.
+	// In IPv6-only mode, calico-node will be rendered with CALICO_ROUTER_ID="hash".
+	ip6, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "IP6")
+	if err != nil {
+		return err
+	}
+	ip, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "IP")
+	if err != nil {
+		return err
+	}
+	if operatorV6Pool.CIDR != "" && ip6 != nil && *ip6 == "autodetect" && ip != nil && *ip == "none" {
+		encap := operatorV6Pool.Encapsulation
+		if encap != operatorv1.EncapsulationVXLAN && encap != operatorv1.EncapsulationVXLANCrossSubnet {
+			if err := c.node.assertEnv(ctx, c.client, containerCalicoNode, "CALICO_ROUTER_ID", "hash"); err != nil {
```

Comment:
Worth noting that just because VXLAN is being used doesn't mean that BGP is going to be disabled. IIRC the operator has a config option for enabling / disabling BGP which is independent of encap.

Reply:
I think that means we need to check the `CALICO_DISABLE_BGP` env var here, not the VXLAN env vars.

Reply:
I fixed this and moved the logic back to `network.go`